plan/new_test.go

```diff
@@ -3,45 +3,186 @@ package plan
 import (
 	"testing"
 
+	"github.com/bradfitz/slice"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestNew(t *testing.T) {
 	plan := New(&Config{
-		Clients: []string{"alpha", "omega"},
-		Axes: []Axis{{
-			Name:   "behavior",
-			Values: []string{"dance", "sing"},
-		}},
+		Waiters: []string{"alpha", "omega"},
+		Axes: map[string]Axis{
+			"client":    {Name: "client", Values: []string{"alpha", "omega"}},
+			"server":    {Name: "server", Values: []string{"alpha", "omega"}},
+			"transport": {Name: "transport", Values: []string{"http", "tchannel"}},
+		},
+		Behaviors: map[string]Behavior{
+			"dance": {Name: "dance",
+				Clients: "client",
+				Params:  []string{"server", "transport"},
+			},
+			"sing": {Name: "sing",
+				Clients: "client",
+				Params:  []string{"server", "transport"},
+			}}})
+
+	slice.Sort(plan.TestCases, func(i, j int) bool {
+		if plan.TestCases[i].Client != plan.TestCases[j].Client {
+			return plan.TestCases[i].Client < plan.TestCases[j].Client
+		}
+		if plan.TestCases[i].Arguments["server"] != plan.TestCases[j].Arguments["server"] {
+			return plan.TestCases[i].Arguments["server"] < plan.TestCases[j].Arguments["server"]
```

Comment:
you can also DRY up this by doing
```go
if diff := p1.Arguments["server"] - p2.Arguments["server"]; diff != 0 {
  return diff < 0
}
```
you can do the same for every test you have.